ui/pages/confirmations/components/confirm/info/hooks/known-method-data-in-transaction.ts

```diff
@@ -0,0 +1,22 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  getKnownMethodData,
+  use4ByteResolutionSelector,
+} from '../../../../../../selectors';
+import { getContractMethodData } from '../../../../../../store/actions';
+
+export const useKnownMethodDataInTransaction = (
+  currentConfirmation: TransactionMeta,
+) => {
+  const dispatch = useDispatch();
+  const use4ByteResolution = useSelector(use4ByteResolutionSelector);
+  if (use4ByteResolution && currentConfirmation?.txParams?.data) {
+    dispatch(getContractMethodData(currentConfirmation.txParams.data));
+  }
+  const knownMethodData =
+    useSelector((state) =>
+      getKnownMethodData(state, currentConfirmation?.txParams?.data),
+    ) || {};
+  return { knownMethodData };
+};
```

Comment:
nit: unit test coverage for the hook will be nice

Reply:
Because `useKnownMethodDataInTransaction`, which is further down the component chain needs to be able to dispatch actions.

Reply:
I added coverage to this hook, thanks for the suggestion!